internal/resources/private_endpoints.go

```diff
@@ -0,0 +1,317 @@
+package resources
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api"
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/errors"
+	providerschema "github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/schema"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &PrivateEndpoint{}
+	_ resource.ResourceWithConfigure   = &PrivateEndpoint{}
+	_ resource.ResourceWithImportState = &PrivateEndpoint{}
+)
+
+type PrivateEndpoint struct {
+	*providerschema.Data
+}
+
+func NewPrivateEndpoint() resource.Resource {
+	return &PrivateEndpoint{}
+}
+
+func (p *PrivateEndpoint) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_private_endpoints"
```

Comment:
Is there any specific reason why we have _private_endpoints instead of _private_endpoint like other resources?

Reply:
My Question Is 

Is there any specific reason you chose _private_endpoints as the resource type name instead of private_endpoint, like we have _cluster, _bucket, and _apikey, none of which are plural?